### PR TITLE
feat: Switch from edx-sphinx-theme to sphinx-book-theme

### PR DESCRIPTION
**Pull request by xitij2000**
The edx-sphinx theme is being deprecated, and replaced with sphinx-book-theme. This removes references to the deprecated theme and replaces them with the new standard theme for the platform.

It also removes some unnecessary overrides from the subtemplates.

*Testing Steps:*

- Follow the project's instructions to create the different types of projects. 
- Ensure that the projects created use the new Sphinx theme, and don't contain other differences that could break things. 

Here is one rough process:

- Checkout the master branch. 
- Run a command like `make cookiecutter-django-app`
- Keep the default name or provide a name for the app (for eg. `app`)
- The project files will be rendered in the var directory (for eg. var/app)
- Rename that folder to something like `app.old`
- Checkout the branch from this PR and repeat the above steps and use the same app name. 
- Using the same app name means that it should create the same structure, and the only differences should be from the change in templates in this PR. 
- Look at the diff and make sure it aligns with the goals of this PR. 
- Repeat for all templates (cookiecutter-django-ida, cookiecutter-python-library, cookiecutter-xblock).

ref: https://github.com/openedx/edx-sphinx-theme/issues/193

**Merge checklist:**
Check off if complete *or* not applicable:
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Fixup commits are squashed away
- [x] Unit tests added/updated
- [x] Manual testing instructions provided
- [x] Noted any: Concerns, dependencies, deadlines, tickets

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @xitij2000!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: cookiecutter-django-app/{{cookiecutter.repo_name}}/docs/conf.py
Comment (xitij2000) [inline cookiecutter-django-app/{{cookiecutter.repo_name}}/docs/conf.py:1]: Removing this file in favour of updating the common conf.py file in `python-template`

File: cookiecutter-django-app/{{cookiecutter.repo_name}}/requirements/doc.in
Comment (xitij2000) [inline cookiecutter-django-app/{{cookiecutter.repo_name}}/requirements/doc.in:1]: Removed in favour of file in `python-template` which this doesn't really alter.

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/requirements/doc.in
Comment (xitij2000) [inline cookiecutter-django-ida/{{cookiecutter.repo_name}}/requirements/doc.in:1]: Removed in favour of file in `python-template` which this doesn't really alter.

**Comment by Cup0fCoffee** [comment]
:+1:

- [x] I tested this
- [x] I read through the code
- [x] Includes documentation

**Comment by itsjeyd** [comment]
Thanks @xitij2000!

Hey @feanil, will you be reviewing this PR, or should I ping someone else from tCRIL?

**Comment by itsjeyd** [comment]
@feanil Friendly nudge about my comment above.

**Comment by feanil** [comment]
Yep, I'll be reviewing this and the other docs PRs from @xitij2000

You can assign them all to me.

**Comment by itsjeyd** [comment]
> Yep, I'll be reviewing this and the other docs PRs from @xitij2000
> 
> You can assign them all to me.

@feanil Great, will do!

File: python-template/{{cookiecutter.placeholder_repo_name}}/docs/concepts.rst
[the remark this answers is not in the document]
Comment (xitij2000) [reply python-template/{{cookiecutter.placeholder_repo_name}}/docs/concepts.rst:1]: Sure, I've removed this and other redundant index files.

File: python-template/{{cookiecutter.placeholder_repo_name}}/docs/conf.py [hunk 1]
```diff
@@ -173,22 +180,52 @@ def get_version(*file_paths):
...
+                src="https://i.creativecommons.org/l/by-sa/4.0/80x15.png"/>
+        </a>
+        <br>
+        These works by
+            <a
+                xmlns:cc="https://creativecommons.org/ns#"
+                href="https://openedx.org"
+                property="cc:attributionName"
+                rel="cc:attributionURL"
+            >The Center for Reimagining Learning</a>
```
Comment (feanil) [inline python-template/{{cookiecutter.placeholder_repo_name}}/docs/conf.py:218]:
```suggestion
            >The Axim Collaborative</a>
```

**Review by feanil — COMMENTED** [review]
1 small change, and also it looks like `make upgrade` needs to be run with python 3.8 instead of python 3.10

Once those two things are done, I think this is good to land.

**Comment by xitij2000** [comment]
> 1 small change, and also it looks like `make upgrade` needs to be run with python 3.8 instead of python 3.10
> 
> Once those two things are done, I think this is good to land.

Rerun upgrade with Python 3.8

File: python-template/{{cookiecutter.placeholder_repo_name}}/docs/conf.py (hunk 1 above)
Comment (feanil) [inline python-template/{{cookiecutter.placeholder_repo_name}}/docs/conf.py:218]:
```suggestion
            >the Axim Collaborative</a>
```

**Review by feanil — COMMENTED** [review]
Looks like the `tCRIL` -> `Axim` update didn't happen.  Also, it looks like this needs to be rebased to deal with some conflicts.

File: python-template/{{cookiecutter.placeholder_repo_name}}/docs/conf.py (hunk 1 above)
Comment (xitij2000) [reply python-template/{{cookiecutter.placeholder_repo_name}}/docs/conf.py]: Sorry, I think I accidentally dropped that commit during rebase. Fixed.

File: python-template/{{cookiecutter.placeholder_repo_name}}/docs/conf.py [hunk 2]
```diff
@@ -90,17 +98,16 @@ def get_version(*file_paths):
 
 # General information about the project.
 project = '{{ cookiecutter.project_name }}'
-copyright = edx_theme.COPYRIGHT  # pylint: disable=redefined-builtin
-author = edx_theme.AUTHOR
+copyright = f'{datetime.now().year}, edX Inc.'  # pylint: disable=redefined-builtin
```
Comment (feanil) [inline python-template/{{cookiecutter.placeholder_repo_name}}/docs/conf.py:101]:
```suggestion
copyright = f'{datetime.now().year}, Axim Collaborative Inc.'  # pylint: disable=redefined-builtin
```

**Review by feanil — COMMENTED** [review]
@xitij2000 sorry 3 more small suggestions to merge and then I think this is good to go.

**Review by feanil — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@xitij2000 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-cookiecutters&entry.867055334=https://github.com/openedx/edx-cookiecutters/pull/305&entry.1484655318=https://github.com/xitij2000&entry.752974735=2023-03-21+10:24&entry.1917517419=2023-04-20+18:16&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by itsjeyd** [comment]
label: core contributor

**Pull request merged**